### PR TITLE
feat: [ANDROAPP-2979] TEI's events that have geometry are now displayed in the map.

### PR DESCRIPTION
**Pull request by ferdyrod**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ANDROAPP-2979](https://jira.dhis2.org/browse/ANDROAPP-2979)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    private val mapPointToFeature: MapPointToFeature,
+    private val mapPolygonToFeature: MapPolygonToFeature
+) {
+
+    fun map(
+        events: List<EventUiComponentModel>
+    ): Pair<HashMap<String, FeatureCollection>, BoundingBox> {
+        val featureMap: HashMap<String, ArrayList<Feature>> = HashMap()
+        featureMap[EVENT] = ArrayList()
+        bounds.initOrReset()
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt:24]: I'm doing a PR so we can remove this bounds thing and use "getBoudingBox"

File: app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        featureCollectionMap[EVENT] = FeatureCollection.fromFeatures(featureMap[EVENT] as ArrayList)
+
+        return Pair<HashMap<String, FeatureCollection>, BoundingBox>(
+            featureCollectionMap,
+            BoundingBox.fromLngLats(
+                bounds.westBound,
+                bounds.southBound,
+                bounds.eastBound,
+                bounds.northBound
+            )
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt:49]: This bounds will be zero all of them. You can use the same one from relationships :)

Comment (JaimeToca) [reply app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt:49]: val latLongList = getLngLatAsCoordinateList(featuresList)
bounds.getEnclosingBoundingBox(latLongList)

File: app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                if (it.type() == FeatureType.POINT) {
+                    mapPointToFeature.map(it)
+                } else {
+                    mapPolygonToFeature.map(it)
+                }
+            }
+            feature?.addTeiEventInfo(eventModel)?.also {
+                featureMap[EVENT]!!.add(it)
+            }
+        }
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt:37]: do you need to classify per program stage?

File: app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/FeatureCollectionExtensions.kt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package org.dhis2.uicomponents.map.geometry.mapper.featurecollection
+
+import com.mapbox.geojson.FeatureCollection
+
+fun HashMap<String, FeatureCollection>.appendFeatureCollection(
+    collectionToAppend: Map<String, FeatureCollection>
+): HashMap<String, FeatureCollection> {
+    this.putAll(collectionToAppend)
+    return this
+}
```
[the remark this answers is not in the document]
Comment (ferdyrod) [reply app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/FeatureCollectionExtensions.kt:10]: This will be removed.

**Comment by JaimeToca** [comment]
good job @ferdyrod :medal_sports:

File: app/src/main/java/org/dhis2/uicomponents/map/managers/TeiMapManager.kt [hunk 5]
```diff
@@ -15,26 +16,38 @@ class TeiMapManager(
...
                 (style?.getSource(ENROLLMENT_SOURCE_ID) as GeoJsonSource?)
                     ?.setGeoJson(teiFeatureCollections[ENROLLMENT])
+            }
+            .also {
+                this.eventsFeatureCollection.apply {
+                    keys.forEach { key ->
+                        (style?.getSource(key) as GeoJsonSource?)
+                            ?.setGeoJson(this[key])
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (ferdyrod) [reply app/src/main/java/org/dhis2/uicomponents/map/managers/TeiMapManager.kt:50]: It is my understanding that if there is a problem getting source/setting the geojson we should call loadDataForStyle, Am I not understanding correctly??

File: app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+        val featureCollectionMap = eventsByProgramStage.mapValues {
+            FeatureCollection.fromFeatures(it.value)
+        }
+
+        val featureCollection = EventsByProgramStage(EVENT, featureCollectionMap)
+
+        val latLngList = eventsByProgramStage.values.flatten()
+            .filter { it?.geometry() is Point }
+            .map { it.getPointLatLng() }
```
[the remark this answers is not in the document]
Comment (ferdyrod) [reply app/src/main/java/org/dhis2/uicomponents/map/geometry/mapper/featurecollection/MapTeiEventsToFeatureCollection.kt:47]: done

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java [hunk 7]
```diff
@@ -243,12 +259,32 @@ public void init(String trackedEntityType) {
...
+                                        .map(teisAndEvents -> new Triple<>(teisAndEvents.component1(),
+                                                mapTeisToFeatureCollection.map(teisAndEvents.component1()),
+                                                mapTeiEventsToFeatureCollection.map(teisAndEvents.component2()))
+                                        )
+                                .map(triple -> {
+                                    HashMap<String, FeatureCollection> teis = triple.component2().component1();
+                                    EventsByProgramStage events = triple.component3().component1();
+                                    kotlin.Triple<HashMap<String, FeatureCollection>, BoundingBox, EventsByProgramStage> feat = new kotlin.Triple<>(teis, triple.component2().component2(), events);
+                                    return new kotlin.Pair<>(triple.component1(), feat);
+                                })
```
[the remark this answers is not in the document]
Comment (ferdyrod) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java:277]: done

File: app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java [hunk 8]
```diff
@@ -597,6 +601,32 @@ else if (d2.trackedEntityModule().trackedEntityAttributes().byUid().eq(tetAttrib
...
+                    .uid(event.programStage())
+                    .blockingGet();
+
+            eventViewModels.add(new EventViewModel(EventViewModelType.EVENT, stage, event, 0, null, true, true));
+        }
+
+        return eventViewModels;
+    }
+
+    // Private Region Start//
```
[the remark this answers is not in the document]
Comment (ferdyrod) [reply app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java:639]: done

File: app/src/test/java/org/dhis2/uicomponents/map/geometry/MapTeiEventsToFeatureCollectionTest.kt [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        val event = EventUiComponentModel(
+            "stageUid",
+            "stageDisplayName",
+            "eventUid",
+            Event.builder().uid(EVENTUID).geometry(
+                Geometry.builder()
+                    .type(FeatureType.POINT)
+                    .coordinates("[$POINT_LONGITUDE,$POINT_LATITUDE]")
+                    .build()
+            ).build(),
```
[the remark this answers is not in the document]
Comment (ferdyrod) [reply app/src/test/java/org/dhis2/uicomponents/map/geometry/MapTeiEventsToFeatureCollectionTest.kt:72]: great, did not know about this. Done!

**Pull request merged**